packages/core/src/components/menu/menuItem.tsx

```diff
@@ -77,6 +77,11 @@ export interface IMenuItemProps extends ActionProps, LinkProps {
      */
     labelElement?: React.ReactNode;
 
+    /**
+     * Props to spread to the child `Menu` component if this item has a submenu.
+     */
+    menuProps?: Partial<MenuProps> & object;
```

Comment:
actually, it's an implementation detail, so it should be a code comment for devs rather than JSDoc:

```suggestion
     */
     // note that `& object` is required to allow attributes built-in to JSX like "data-testid"
    submenuProps?: Partial<MenuProps> & object;
```

Reply:
I'm just going to remove the `& object`, that actually didn't solve it. Just  ` Partial<MenuProps>` is fine.